metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/EnrichmentWorkerBuilder.java

```diff
@@ -113,34 +43,12 @@ public EnrichmentWorkerBuilder setBatchSizeEnrichment(int batchSizeEnrichment) {
   public EnrichmentWorker build() {
 
     // Make sure that the worker can do something.
-    if (StringUtils.isBlank(dereferenceUrl) && StringUtils.isBlank(enrichmentUrl)) {
+    if (dereferencer == null && enricher == null) {
       throw new IllegalStateException(
               "Either dereferencing or enrichment (or both) must be enabled.");
     }
 
-    // Create the request factory
-    final HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
-    requestFactory.setConnectTimeout(Math.max(connectTimeout, 0));
-    requestFactory.setReadTimeout(Math.max(responseTimeout, 0));
-
-    // Create the dereference client if needed
-    final DereferenceClient dereferenceClient;
-    if (StringUtils.isNotBlank(dereferenceUrl)) {
-      dereferenceClient = new DereferenceClient(new RestTemplate(requestFactory), dereferenceUrl);
-    } else {
-      dereferenceClient = null;
-    }
-
-    // Create the enrichment client if needed
-    final EnrichmentClient enrichmentClient;
-    if (StringUtils.isNotBlank(enrichmentUrl)) {
-      enrichmentClient = new EnrichmentClient(new RestTemplate(requestFactory), enrichmentUrl,
-              batchSizeEnrichment);
-    } else {
-      enrichmentClient = null;
-    }
-
     // Done.
-    return new EnrichmentWorkerImpl(dereferenceClient, enrichmentClient, new EntityMergeEngine());
+    return new EnrichmentWorkerImpl(dereferencer, enricher);
```

Comment:
This contradicts the Builder pattern where the user can actually create this object by using this constructor(normally the constructor should be private)..
Are we allowing for that?
If yes then perhaps the Builder naming should change to something else to avoid confusion..

(This looks more like a factory pattern to me)

Reply:
Now that you bring this up, it made me realize that maybe we don't even need a builder for this class, since it is basically the same as using the original constructor.
I will delete this `Builder` class